File: p1/1-Trabalho/seidel.c
#include <stdio.h>
#include <math.h>
#define L 3
#define C 3

void jacobi(double A[L][C], double B[L], double chute[L], int n){
    for(int k=0;k<n;k++){
        for(int i=0;i<L;i++){
            double bi=B[i];
            for(int j=0;j<C;j++){
                if(j!=i) bi-=A[i][j]*chute[j];
            }
            bi/=A[i][i];
            printf("x_%d(%d) = %.16f | ", i+1, k+1, bi);
            chute[i]=bi;
        }
        printf("\n");
    }
}


// INSERIR O TAMANHO DA MATRIZ EM DEFINE
//***************************************************************************

int main(){
    double A[L][C]={{0.48,0.17,0.35},{0.19,0.51,0.30},{0.23,0.18,0.59}};
    double B[L]={4094,3311,5496};

    double chute[L]= {1000,500,500};
    int n=16;

    jacobi(A, B, chute, n);

    return 0;
}

File: p1/sistema linear/seidel.py
from jacobi import metodoJacobi
import math

#   SISTEMAS LINEARES
#           METODO DA ELIMINACAO DE GAUSS SEIDEL
# Entrada: matriz[lin][col], termosIndependentes[col], chute[col], numIteracao
# matriz é o sistema linear
# TI são as respostas em vetor do das equacoes do SL

def metodoGaussSeidel(matriz,ti,chute,numIteracao):
  linha=len(matriz)
  coluna=len(matriz[0])
  for it in range(numIteracao):
    for i in range(linha):
      xi = ti[i]
      for j in range(coluna):
        if i != j:
          xi -= matriz[i][j] * chute[j]
      xi /= matriz[i][i]
      chute[i] = xi
    #if it == 2 or it == 8 or it == 16:
    print("X^{} -> ".format(it+1))
    for k in range(coluna):
      print("{}".format(chute[k]))
    print("")

if __name__ == "__main__":
  A = [[1, -1,0], [-45.82,-62.28,-75.75], [0,1,-1]]
  B = [- 318.2694, - 510.42960000000005, - 683.2089000000001]
  chute = [0,0,0]
  n=1

  metodoGaussSeidel(A,B,chute,n)
